src/geom/SipApproximation.cc

```diff
@@ -264,24 +264,24 @@ double SipApproximation::getA(int p, int q) const {
 }
 
 double SipApproximation::getB(int p, int q) const {
-    return _solution->b[_solution->a.getBasis().index(p, q)];
+    return _solution->b[_solution->b.getBasis().index(p, q)];
 }
 
 double SipApproximation::getAP(int p, int q) const {
-    return _solution->ap[_solution->a.getBasis().index(p, q)];
+    return _solution->ap[_solution->ap.getBasis().index(p, q)];
 }
 
 double SipApproximation::getBP(int p, int q) const {
-    return _solution->bp[_solution->a.getBasis().index(p, q)];
+    return _solution->bp[_solution->bp.getBasis().index(p, q)];
```

Comment:
Yikes!  Glad you caught these.  I wonder what I was doing in my tests that made it going unnoticed.  Maybe just unlucky float values that were all close to zero?